makefiles/deprecated_modules.inc.mk

```diff
@@ -1,5 +1,31 @@
 # Add deprecated modules here
-# Keep this list ALPHABETICALLY SORTED!!!!111elven
+# Keep this list ALPHABETICALLY SORTED!!!!
```

Comment:
I'm a bit skeptical about this warning. As a user, I suddenly get a warning on a module that I've been using so far. It says that the module behaviour will change, but provides no update path in which one can get rid of the warning. I don't really want to bikeshed this cleanup, but I'm afraid it may be a bit confusing for users. An alternative would be to provide `at24mac_id` to pull in only the ID functionality, but it would not follow the convention of the module namespacing.

@benpicco do you have an opinion on this?

Reply:
I added an `I_UNDERSTAND_AT24MAC_WILL_ONLY_BE_ID` flag to prevent the warnings until the deprecation is finished.

I don't like adding the `_id` as this would have to last, creating a difference between the actual module (folder, header, namespacing) and the USEMODULE. Even though that is "allowed" in RIOT I would like to keep things as standard as possible.

Reply:
I really don't understand why all the hassle is needed just to avoid compiling a 89 line `.c` file.

Reply:
Done